### ISSUE TITLE
Missing timepicker component

### ISSUE THREAD

**Issue by skateman** [data-driven-forms/react-forms#103]
This component is required for the service dialog editor/user, in the old world it looks like this:
![Screenshot from 2019-09-05 09-57-08](https://user-images.githubusercontent.com/649130/64322838-9660a800-cfc3-11e9-8a11-9ae2499c4112.png)

So basically a datepicker with a time picker.

**Comment by Hyperkid123** [comment]
@skateman there is data-time picker but is not being shown in docs(we had no requirement for it). I will add it to the docs.

**Comment by skateman** [comment]
@Hyperkid123 is it possible to alias it to `TIME_PICKER`?

**Comment by Hyperkid123** [comment]
Sure

**Comment by Hyperkid123** [comment]
@skateman we added a variant prop to date picker so now you have the date and time picker in the popover according to PF3 design. https://data-driven-forms.org/component-example/date-picker

We will not be calling it time picker because PF does actually have a time picker design (just the time part without the calendar) which will have the constant `TIME_PICKER` once we add it. As of this moment we are not sure how the API should look like for time picker component so it is not exposed just yet. We first need to see what values would be expected from a component like that.

Is that good enough or do you actually need the time picker not the date-time picker?

**Comment by skateman** [comment]
In the MiQ dialog editor the time picker is actually a datetime picker. If you think it through, there is no reason in the context of MiQ to have a time picker that only selects time and not data. I'm pretty sure that this is the case of any product we will be integrating DDF into, but I might be wrong. Anyway, I would need at least a `DATETIME_PICKER`. Having a separate component for date and datetime actually makes sense, even if you're using the same component for rendering it.

**Comment by Hyperkid123** [comment]
@skateman  The fact that the MIQ dialog editor (or any consuming app/package) requires an alias should not be good enough argument to add it. If we start doing that we could end up with fifty different aliases for text input. That is not something we want to happen. However there is an option if you want to have an aliases for any components from mappers 😉.
```jsx
import { components, formFieldsMapper, layoutMapper } from '@data-driven-forms/pf3-component-mapper';
import FormRenderer from '@data-driven-forms/react-form-renderer';

const schema = {fields: [
  {
    component: 'alias',
    label: 'Foo',
    name: 'alias-date-picker',
    variant: 'this-prop-does-not-matter' // is overridden in custom formFieldsMapper
  }
]}

function App() {
  console.log(components.DatePickerField)
  const foo = {
    ...formFieldsMapper,
    'alias': ({FieldProvider, ...props}) => <FieldProvider component={components.DatePickerField} {...props} variant="date-time" />
  }
  return (
    <div>
      <FormRenderer schema={schema} onSubmit={console.log} formFieldsMapper={foo} layoutMapper={layoutMapper} />
    </div>
  );
}
```

You can use the component you need give it another name and override the props (variant in this case) to match exactly what you need.

Is this solution acceptable?

**Comment by skateman** [comment]
Well, I'm okay with a component mapper alias, but then we will have to implement it MiQ separately. However, as far as I understood, at the beginning the DDF was trying to implement the components from the dialog editor. I guess that's the reason of the existence of the tag control component. Now you're saying that you no longer want to have this compatibility? 

Even though the date picker and the datetime picker are rendered by the same 3rd party component as a variant, they represent different kind of data types (date vs datetime). IMHO having one component type in our world that produces different data types based on a parameter is a bad practice and I'd like to avoid it if that's possible. So I do see the reason for a separate component type.

On the other hand, I understand your concerns of having too many component types. But following this logic we could just deprecate radio components and set a variant for dropdown to be rendered as radios...

**Comment by Hyperkid123** [comment]
> at the beginning the DDF was trying to implement the components from the dialog editor. 

That is not true it started as a component for insights which was to render forms based on OpenAPI schemas.

> I guess that's the reason of the existence of the tag control component.

We the made a parser from miq schema to data driven forms and we needed a placeholder so the so the component wont die in runtime. Now that we know that tag selector is just an async select which automatically resolves url, It will be deprecated and we will alias create alias for select.

**Comment by skateman** [comment]
Ok, is there an issue or any other place where these deprecated items are collected?

**Comment by Hyperkid123** [comment]
No i want to start a project for v2.0 where all this changes will take an effect. Let me start it right away.
